src/Components/Web.JS/src/Platform/Mono/MonoPlatform.ts

```diff
@@ -562,6 +564,10 @@ class MonoHeapLock implements HeapLock {
       throw new Error('Trying to release a lock which isn\'t current');
     }
 
+    if (typeof MONO_INTERNAL !== 'undefined' && typeof MONO_INTERNAL.mono_wasm_gc_unlock === 'function') {
```

Comment:
I understand this is a point-in-time thing, but maybe we should wait until the runtime is merged before we complete this PR.

It would be ideal *not* to silently skip `mono_wasm_gc_lock`/`unlock` if they can't be found, because then we'll be at risk of accidentally disabling this functionality in the future if the name changes or similar. It would be preferable to fail hard if this function can't be found.

Reply:
> if it's paired with a runtime that doesn't have the API, it behaves as before.

I'm not aware that we need to support mismatches of runtime and Blazor versions. Please let me know if I'm missing something!

Reply:
Correct, we don't support version mismatch. This is point-in-time comment.

Reply:
If that's the case and we don't want to change it on this PR, can we file an issue to track its removal once the changes are in so that we don't forget?

Reply:
I think the whole PR could wait in current state until the runtime version flows.

Reply:
The runtime changes are in https://github.com/dotnet/aspnetcore/pull/47021

Reply:
Ok, codeflow from runtime is done.  We're unblocked.  Please take a look again

Reply:
Or maybe not.  looks like we also need an SDK bump

Reply:
Or maybe not. looks like we also need an SDK bump

Reply:
I think we're blocked on https://github.com/dotnet/aspnetcore/pull/47062 which shoudl pull in a runtime that includes at least dotnet/runtime@c7537bdf261 which shoudl have the gc lock API.

I should expect to see failing tests, right?

Reply:
Looks like we're good now